### PR TITLE
Merging in Current Web Functionality

### PR DESCRIPTION
**Pull request by Clare2D**
Initially checking the merge and will see what needs to be resolved.

### CODE REVIEW

**Review by Clare2D — COMMENTED** [review]
So I've had a look at all the files and am happy that this merge should be alright to happen. I've checked with Jacob my clarifications about things which have been resolved. Would be keen to have you both look at it anyway to check anything I've missed.

**Comment by Clare2D** [comment]
So while resolving issues I created a new branch which seems to have replaced this main branch. I hope this isn't problematic.

File: 0_fund_processing.R
Comment (jacobvjk) [inline 0_fund_processing.R:3]: I don't really understand what this script is doing in this repo tbh.. not necessarily relevant for this PR, but just saying @Clare2D

File: 0_global_functions.R
Comment (jacobvjk) [inline 0_global_functions.R:24]: This should probably get another name or not be used, as it could mask the readr::read_file() function and lead to unexpected behaviour/errors

File: 0_portfolio_input_check_functions.R
Comment (jacobvjk) [inline 0_portfolio_input_check_functions.R:852]: this will probably cause the masking error @Clare2D I would argue readRDS is safer for now and we should change the function name

File: 0_portfolio_input_check_functions.R
Comment (jacobvjk) [inline 0_portfolio_input_check_functions.R:990]: same issue here. I would argue against read_file in its current form @Clare2D

File: 0_portfolio_input_check_functions.R
Comment (jacobvjk) [inline 0_portfolio_input_check_functions.R:1304]: function masking issue @Clare2D

File: 0_portfolio_input_check_functions.R
Comment (jacobvjk) [inline 0_portfolio_input_check_functions.R:1315]: function see above @Clare2D

File: 5_boe_stress_test.R
Comment (jacobvjk) [inline 5_boe_stress_test.R:1]: this can be removed entirely

File: 0_stress_test_functions.R
Comment (jacobvjk) [inline 0_stress_test_functions.R:1]: this can be removed entirely

**Review by jacobvjk — CHANGES_REQUESTED** [review]
Some function masking issues potentially still in here with read_file instead of read_rds or readRDS.. see comments

**Comment by Clare2D** [comment]
Thanks Jacob. removed those files and I changed the "read_in" function, but I've pushed that to the current web functionality branch.

**Comment by maurolepore** [comment]
Thanks a lot @Clare2D  and @jacobvjk for being open to this change. I know it's a pain but it should pay-off soon.

Git offers some merge strategies that in some cases work well automatically.

Which merge strategy would you say we need?:

1. `current_web_functionaliry` overwrites  `master`.
2. `master` overwrites `current_web_functionality`.
3. We need to solve this manually, file by file.

**Comment by maurolepore** [comment]
> So I've had a look at all the files and am happy that this merge should be alright to happen. I've checked with Jacob my clarifications about things which have been resolved. Would be keen to have you both look at it anyway to check anything I've missed.

-- @Clare2D 

Thanks!

* :+1:: I see [The integration test passes on all platforms and versions of R](https://github.com/2DegreesInvesting/PACTA_analysis/actions/runs/289617141). This gives me confidence that the merged result will continue to work as expected. The user-facing behaviour is appantely preserved, regardless of the details that @jacobvjk comments above. I think it's great to note those details, and maybe flag `# TODO`s and `# FIXME`s, but I lean towards addressing those issues in separate PRs. If the follow-up PRs are ready to go, they can be merged immediately after the merge PR. The resulting history should be clear and easy to revert each step along the way.

* :-1:: I see this branch is not `current_web_functionality`. While merging a clone of `current_web_functionality` should produce the same effect than merging `current_web_functionality` itself, I prefer to merge `current_web_functionality` because it more clearly tells the story of this repository. I suggest not merging this PR -- so I'll change it to a draft to avoid merging it by accident.

**Comment by maurolepore** [comment]
I closed this PR accidentally. Reopening now.

**Review by maurolepore — APPROVED** [review]

**Comment by maurolepore** [comment]
Please see https://github.com/2DegreesInvesting/PACTA_analysis/pull/127. Closing now.

**Pull request closed without merging**
